stl/src/syserror.cpp

```diff
@@ -113,29 +112,59 @@ _CRTIMP2_PURE unsigned long __CLRCALL_PURE_OR_CDECL _Winerror_message(
     unsigned long _Message_id, char* _Narrow, unsigned long _Size) { // convert to name of Windows error, return 0 for
                                                                      // failure, otherwise return number of chars
                                                                      // written pre: _Size < INT_MAX
-    const unique_ptr<wchar_t[]> _Wide(new wchar_t[_Size]); // not using make_unique because we want default-init
-    const auto _First = _Wide.get();
-    unsigned long _Wide_chars =
-        FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, 0, _Message_id, 0, _First, _Size, 0);
+    unsigned long _Chars = FormatMessageA(
+        FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, 0, _Message_id, 0, _Narrow, _Size, 0);
 
     // remove any trailing whitespace
-    while (_Wide_chars != 0) {
-        switch (_First[--_Wide_chars]) {
-        case L' ':
-        case L'\n':
-        case L'\r':
-        case L'\t':
-        case L'\0':
+    while (_Chars != 0) {
+        switch (_Narrow[--_Chars]) {
+        case ' ':
+        case '\n':
+        case '\r':
+        case '\t':
+        case '\0':
             continue;
         default:
-            return static_cast<unsigned long>(
-                WideCharToMultiByte(CP_ACP, 0, _First, _Wide_chars + 1, _Narrow, _Size, 0, 0));
+            return _Chars;
         }
     }
 
     return 0;
 }
 
+_CRTIMP2_PURE size_t __CLRCALL_PURE_OR_CDECL _Winerror_message2(

```

Comment:
I think you need to arrange for the "2" version to be in the import library as we cannot change the export surface of `msvcp140.dll`

Reply:
I'm not quite familiar with the STL's internals so I just copied the _Winerror_message code and adjusted it as required. How would I proceed to do that?

Reply:
If you look in `CMakeLists.txt` we already have a couple .cpps which go in the import lib like `parallel_algorithms.cpp`, you should be able to add a new `.cpp` that goes into the import lib in the same way.

Reply:
I'm looking at this but however I'm not sure how I would share `_TrimEnd` between `_Winerror_message` and `_Winerror_message2` if it's not exported by the DLL.

Reply:
You can't sure the actual assembly / code but you could put the implementation in a shared header. I was pointing out that it should be considered, not saying it is a thing we need to block over.